tests/Functional/Schema/SqliteSchemaManagerTest.php

```diff
@@ -236,4 +236,23 @@ public function testOnlyOwnCommentIsParsed(): void
             ->getColumn('col1')
             ->getComment());
     }
+    
+    public function testListTableIndexesWithIndexExpression(): void
+    {
+        $sql = <<<SQL
+CREATE TABLE company (
+    com_id int(4),
+    com_name text(15)
+);
+CREATE INDEX com_id_index ON company (com_id);
+CREATE INDEX com_name_5_index ON company (substr(com_name, 0, 5))
+SQL;
+
+        $this->connection->executeStatement($sql);
+
+        $indexes = $this->schemaManager->listTableIndexes('company');
+
+        self::assertArrayHasKey('com_id_index', $indexes);
+        self::assertEquals('substr(com_name, 0, 5)', $indexes['com_name_5_index']->getColumns()[0]);
```

Comment:
https://github.com/doctrine/dbal/blob/936e7e3350dd15a2d3de147821dfe5249169ed8b/src/Schema/Constraint.php#L22-L28

`substr(com_name, 0, 5)` isn't a column name. It looks like the `Index` abstraction in the DBAL wasn't designed for exposing column expressions.